lib/dereference.js

```diff
@@ -82,6 +83,11 @@ function crawl(obj, path, pathFromRoot, parents, processedObjects, dereferencedC
           const keyPath = Pointer.join(path, key);
           const keyPathFromRoot = Pointer.join(pathFromRoot, key);
           const value = obj[key];
+
+          if (value?.$ref?.substring(0, 11) === '#components') {
```

Comment:
what about checking if the first character is a `#` and if it's not immediately followed with a `/` then we throw the `InvalidPointerError`?

Reply:
so I think there are some cases where it's _technically_ valid? like in this case here. the `#fragment` at the bottom:
```title: Person
type: object
definitions:
  name:
    type: object
    required:
      - first
      - last
    properties:
      first:
        $ref: "#/definitions/requiredString"
      last:
        $ref: "#/definitions/name/properties/first"
      middle:
        type:
          $ref: "#/definitions/name/properties/first/type"
        minLength:
          $ref: "#/definitions/name/properties/last/minLength"
      prefix:
        $ref: "#/definitions/name/properties/last"
        minLength: 3
      suffix:
        type: string
        $ref: "#/definitions/name/properties/prefix"
        maxLength: 3
  requiredString:
    title: requiredString
    type: string
    minLength: 1
  fragment:
    $id: "#fragment"
required:
  - name
properties:
  name:
    $ref: "#/definitions/name"
  age:
    type: integer
    minimum: 0
  gender:
    type: string
    enum:
      - male
      - female
  fragment:
    $ref: "#fragment"

Reply:
I updated the logic to the following:
```
const splitRef = value?.$ref?.split("/");
          
if (splitRef?.length > 1 && splitRef?.includes("#components")) {
  throw new InvalidPointerError(value.$ref, keyPath);
}
```

if I split the `$ref` up by `/` and there's more than 1 entry, I'm assuming the person has a `$ref` that's pointing to another component/schema. in that case, if it includes `#components`, it throws an error. if there's only one entry, I'm going to assume it's something like `#fragment`

Reply:
Yeah the `#fragment` stuff will only ever be on `$id` props